src/redux/stores/transactions/transactionsSlice.ts

```diff
@@ -1,196 +1,55 @@
-    builder.addCase(submitTransaction, (state, action) => {
-      state.all.unshift(action.payload);
-    });
-    builder.addCase(declineTransaction, (state, action) => {
-      console.error(action.payload);
-      clearExpiry(action.payload.signerWallet, action.payload.nonce);
-      updateTransaction({
-        state,
-        hash: action.payload.hash,
-        nonce: action.payload.nonce,
-        signerWallet: action.payload.signerWallet,
-        status: 'declined',
-        protocol: action.payload.protocol,
-      });
-    });
-    builder.addCase(revertTransaction, (state, action) => {
-      clearExpiry(action.payload.signerWallet, action.payload.nonce);
-      updateTransaction({
-        state,
-        signerWallet: action.payload.signerWallet,
-        nonce: action.payload.nonce,
-        hash: action.payload.hash,
-        status: 'reverted',
-      });
-    });
-    builder.addCase(expireTransaction, (state, action) => {
-      const { signerWallet, nonce } = action.payload;
-      clearExpiry(signerWallet, nonce);
-      updateTransaction({
-        state,
-        signerWallet,
-        nonce,
-        status: 'expired',
-      });
-    });
-    builder.addCase(mineTransaction, (state, action) => {
-      clearExpiry(action.payload.signerWallet, action.payload.nonce);
-      updateTransaction({
-        state,
-        hash: action.payload.hash,
-        nonce: action.payload.nonce,
-        signerWallet: action.payload.signerWallet,
-        status: 'succeeded',
-        protocol: action.payload.protocol,
-      });
-    });
-  },
 });
 
-export const { clear, setTransactions } = transactionsSlice.actions;
-export const selectTransactions = (state: RootState) => state.transactions.all;
-
-export const selectPendingTransactions = createSelector(
-  selectTransactions,
-  (transactions) => transactions.filter((tx) => tx.status === 'processing'),
-);
-
-export const selectPendingDeposits = (
-  state: RootState,
-): SubmittedDepositOrder[] => state.transactions.all.filter(
-  (tx) => tx.status === 'processing' && tx.type === 'Deposit',
-) as SubmittedDepositOrder[];
-
-export const selectApprovals = (state: RootState) => state.transactions.all.filter(
-  (tx) => tx.type === 'Approval',
-) as SubmittedApproval[];
-
-export const selectPendingApprovals = (state: RootState) => state.transactions.all.filter(
-  (tx) => tx.status === 'processing' && tx.type === 'Approval',
-) as SubmittedApproval[];
-
-export const selectPendingOrders = (state: RootState) => state.transactions.all.filter(
-  (tx) => tx.status === 'processing' && tx.type === 'Order',
-) as SubmittedOrder[];
+export const {
+  setIsLoading,
+  setTransactions,
+} = transactionsSlice.actions;
 
-export const selectOrders = (state: RootState) => state.transactions.all.filter(
-  (tx) => tx.type === 'Order',
-) as SubmittedOrder[];
+export const selectErc20ApprovalTransactions = (state: RootState): Erc20ApprovalTransaction[] => state.transactions.transactions
```

Comment:
I don't know if it could be safer for selectors to chain ? if the path does not exist in store 
What do you think ?
state?.transactions?.transactions?.filter.?(isErc20ApprovalTransaction);

Reply:
I'm sure that this path always exists so this is not necessary.